### PR TITLE
dotMorten lib is not needed, and this is not a bug fix.

### PR DESCRIPTION
**Pull request by mehdihaghshenas**

### CODE REVIEW

**Comment by ErikEJ** [comment]
This library should use the official Microsoft.SqlServer.Types package.

**Comment by ErikEJ** [comment]
Also EF Core 7 targets .NET 6

**Comment by mehdihaghshenas** [comment]
> This library should use the official Microsoft.SqlServer.Types package.

but https://www.nuget.org/packages/Microsoft.SqlServer.Types/ only support dot net 4.x

**Comment by ErikEJ** [comment]
No, the latest version supports .NET Standard 2.1 https://www.nuget.org/packages/Microsoft.SqlServer.Types/160.900.6-rc0#supportedframeworks-body-tab

**Comment by mehdihaghshenas** [comment]
> No, the latest version supports .NET Standard 2.1 https://www.nuget.org/packages/Microsoft.SqlServer.Types/160.900.6-rc0#supportedframeworks-body-tab

Thank you so much

**Comment by mehdihaghshenas** [comment]
@bricelam 
can you publish a new version of this package
we have problem to use `rc` version with `EF 7.0.0`

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     - name: Setup .NET
       uses: actions/setup-dotnet@v2
       with:
-        dotnet-version: 7.0.100-rc.2.22477.23
+        dotnet-version: 7.0.100-rc.1.22431.12
```
Comment (ErikEJ) [inline .github/workflows/dotnet.yml:21]: Wrong SDK version

File: EFCore.SqlServer.HierarchyId.Abstractions/EFCore.SqlServer.HierarchyId.Abstractions.csproj [hunk 2]
```diff
@@ -10,7 +10,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="dotMorten.Microsoft.SqlServer.Types" Version="1.5.0" />
+	  <PackageReference Include="Microsoft.SqlServer.Types" Version="160.900.6-rc0" />
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Abstractions/EFCore.SqlServer.HierarchyId.Abstractions.csproj:13]: Nit - indentation

File: EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj [hunk 3]
```diff
@@ -8,11 +8,13 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.3.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational.Specification.Tests" Version="7.0.0-rc.2.22472.11" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="7.0.0-rc.2.22472.11" />
-    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="7.0.0-rc.2.22472.3" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.4.0-4.final" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="7.0.0">
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj:12]: Add closing tag

File: EFCore.SqlServer.HierarchyId.Test/RelationalScaffoldingModelFactoryTest.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ namespace Microsoft.EntityFrameworkCore.SqlServer;
 
 public class RelationalScaffoldingModelFactoryTest
 {
-    private readonly IScaffoldingModelFactory _factory;
+    private readonly Microsoft.EntityFrameworkCore.Scaffolding.IScaffoldingModelFactory _factory;
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/RelationalScaffoldingModelFactoryTest.cs:15]: Add a using?

File: EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/AnonymousArraySeedContext.cs [hunk 5]
```diff
@@ -43,11 +43,11 @@ public partial class {migrationName} : Migration
         protected override void Up(MigrationBuilder migrationBuilder)
         {{
             migrationBuilder.CreateTable(
-                name: ""{nameof(ConvertedTestModels)}"",
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/AnonymousArraySeedContext.cs:46]: Why are these test changes needed?

**Comment by ErikEJ** [comment]
Suggest you update the title and initial comment on this PR - dotMorten lib is not needed, and this is not a bug fix.

File: EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/AnonymousArraySeedContext.cs (hunk 5 above)
Comment (mehdihaghshenas) [reply EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/AnonymousArraySeedContext.cs:46]: I changed it to your advice

File: EFCore.SqlServer.HierarchyId.Abstractions/EFCore.SqlServer.HierarchyId.Abstractions.csproj (hunk 2 above)
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Abstractions/EFCore.SqlServer.HierarchyId.Abstractions.csproj:13]: Still indented

File: EFCore.InMemory.HierarchyId/EFCore.InMemory.HierarchyId.csproj [hunk 6]
```diff
@@ -12,7 +12,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="7.0.0-rc.2.22472.11" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="7.0.0" />
+
```
Comment (ErikEJ) [inline EFCore.InMemory.HierarchyId/EFCore.InMemory.HierarchyId.csproj:16]: remove empty line

File: EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj [hunk 7]
```diff
@@ -1,24 +1,25 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
-  <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
-    <IsPackable>false</IsPackable>
-    <AssemblyName>EntityFrameworkCore.SqlServer.HierarchyId.Test</AssemblyName>
-    <RootNamespace>Microsoft.EntityFrameworkCore.SqlServer</RootNamespace>
-  </PropertyGroup>
+	<PropertyGroup>
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj:3]: Fix indentation

File: EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj [hunk 8]
```diff
@@ -1,24 +1,25 @@
...
-  <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.3.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational.Specification.Tests" Version="7.0.0-rc.2.22472.11" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="7.0.0-rc.2.22472.11" />
-    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="7.0.0-rc.2.22472.3" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
-    <PackageReference Include="xunit" Version="2.4.2" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.5" />
-  </ItemGroup>
+	<ItemGroup>
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/EFCore.SqlServer.HierarchyId.Test.csproj:10]: Fix indentation (probably use spaces)

File: EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/AnonymousArraySeedContext.cs [hunk 9]
```diff
@@ -47,7 +47,7 @@ protected override void Up(MigrationBuilder migrationBuilder)
                 columns: table => new
                 {{
                     Id = table.Column<int>(type: ""int"", nullable: false),
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/AnonymousArraySeedContext.cs:49]: Still unneeded test changes !?

File: EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/TypedArraySeedContext.cs [hunk 10]
```diff
@@ -47,7 +47,7 @@ protected override void Up(MigrationBuilder migrationBuilder)
                 columns: table => new
                 {{
                     {nameof(ConvertedPatriarch.Id)} = table.Column<int>(type: ""int"", nullable: false),
-                    {nameof(ConvertedPatriarch.HierarchyId)} = table.Column<{nameof(HierarchyId)}>(type: ""hierarchyid"", nullable: true),
```
Comment (ErikEJ) [inline EFCore.SqlServer.HierarchyId.Test/Test/Models/Migrations/TypedArraySeedContext.cs:50]: Unneeded changes ?

**Comment by bricelam** [comment]
New versions have been uploaded to NuGet.org. See #113 for switching to the official SqlServer.Types package

**Pull request closed without merging**